### ISSUE TITLE
[Bug?]: Adding a portal from a package outside a workspace to one inside fails in 3.2.0

### ISSUE THREAD

**Issue by spiltcoffee** [yarnpkg/berry#4160]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

I have the following package structure:
* `some-workspace`
  * `foo`
  * `bar`
* `baz`

The dependencies of these packages are as such:
* `some-workspace` has a `workspaces` key listing `foo` and `bar`
* `bar` has a `dependencies` entry for `"foo": "workspace:*"`
* `baz` has a `dependencies` entry for `"bar": "portal:../some-workspace/bar"`
* `baz` has a `resolutions` entry for `"bar/foo": "portal:../some-workspace/foo"`

Up to Yarn v3.1.1, this setup worked fine - no errors, no real funkiness.

With Yarn v3.2.0, when I run `yarn install` inside `baz`, I'm now seeing an error occur with this kind of setup, along these lines:
```
➤ YN0001: │ Error: Assertion failed: Writing attempt prevented to /some-workspace/bar/node_modules/foo which is outside project root: /baz
```

### To reproduce

I've created a reproduction here: https://github.com/spiltcoffee/yarn-workspace-portal-issue

### Environment

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz
  Binaries:
    Node: 16.13.2 - C:\Temp\xfs-82221d88\node.CMD
    Yarn: 3.2.0 - C:\Temp\xfs-82221d88\yarn.CMD
    npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
```

### Additional context

I've also tried the following in an attempt to fix the issue:
* Adding a dependency on `foo` that has the same `portal:`
* Adding a `packageExtensions` key to the `.yarnrc.yml`
* Updated to `canary` (it appears to just be Yarn v3.2.0 at the moment anyway?)

**Comment by IU520CF** [comment]
How to delete yarn3.2 ?

**Comment by spiltcoffee** [comment]
@IU520CF Unsure what you mean? Do you mean how to downgrade to an older version of yarn?

**Comment by IU520CF** [comment]
Yes, downgrade yarn2

**Comment by IU520CF** [comment]
I upgraded yarn3 2. I found that I can't use it. Finally, I started it with NPM. The package is bigger

**Comment by larixer** [comment]
The issue should be fixed via: #4206
Also, you can work around it by adding `nmSelfReferences: false` into `.yarnrc.yml`
